lib/file_transfer_agent/file_transfer_agent.js

```diff
@@ -0,0 +1,713 @@
+  {
+    if (stageLocationType === S3_FS)
+    {
+      var client = SnowflakeRemoteStorageUtil.createClient(stageInfo, false);
+      var s3location = SnowflakeS3Util.extractBucketNameAndPath(stageInfo['location']);
+
+      await client.getBucketAccelerateConfiguration({ Bucket: s3location.bucketName })
+        .promise()
+        .then(function (data)
+        {
+          useAccelerateEndpoint = data['Status'] == 'Enabled';
+        }).catch(function (err)
+        {
+          if (err['code'] === 'AccessDenied')
+          {
+            return;
+          }
+        });
+    }
+  }
+
+  /**
+  * Update presigned URLs of file metadata when using GCS client.
+  *
+  * @returns {null}
+  */
+  async function updateFileMetasWithPresignedUrl()
+  {
+    var storageClient = getStorageClient(stageLocationType);
+
+    // presigned url only applies to remote storage
+    if (storageClient === SnowflakeRemoteStorageUtil)
+    {
+      // presigned url only applies to GCS
+      if (stageLocationType === GCS_FS)
+      {
+        if (commandType == CMD_TYPE_UPLOAD)
+        {
+          var filePathToReplace = getFileNameFromPutCommand(command);
+
+          for (var meta of fileMetadata)
+          {
+            var fileNameToReplaceWith = meta['dstFileName'];
+            var commandWithSingleFile = command;
+            var commandWithSingleFile = commandWithSingleFile.replace(filePathToReplace, fileNameToReplaceWith);
+
+            var options = { sqlText: commandWithSingleFile };
+            var newContext = statement.createContext(options, context.services, context.connectionConfig);
+
+            var ret = await statement.sendRequest(newContext);
+            meta['stageInfo'] = ret['data']['data']['stageInfo']
+            meta['presignedUrl'] = meta['stageInfo']['presignedUrl'];
+          }
+        }
+        else if (commandType == CMD_TYPE_DOWNLOAD)
+        {
+          for (var index = 0; index < fileMetadata.length; index++)
+          {
+            if (presignedUrl.length > index)
+            {
+              meta['presignedUrl'] = presignedUrl[index];
+            }
+          }
+        }
+      }
+    }
+  }
+
+  /**
+  * Returns the local file path.
+  *
+  * @param {String} command
+  *
+  * @returns {String}
+  */
+  function getFileNameFromPutCommand(command)
+  {
+    // Extract file path from PUT command:
+    // E.g. "PUT file://C:<path-to-file> @DB.SCHEMA.%TABLE;"
+    var startIndex = command.indexOf(FILE_PROTOCOL) + FILE_PROTOCOL.length;
+    var endIndex = command.substring(startIndex).indexOf(' ');
+    var filePath = command.substring(startIndex, startIndex + endIndex);
+    return filePath;
+  }
+
+  /**
+  * Get the storage client based on stage location type.
+  *
+  * @param {String} stageLocationType
+  *
+  * @returns {Object}
+  */
+  function getStorageClient(stageLocationType)
+  {
+    if (stageLocationType == LOCAL_FS)
+    {
+      return new SnowflakeLocalUtil();
```

Comment:
Uncaught TypeError: SnowflakeLocalUtil is not a constructor
      at getStorageClient (lib/file_transfer_agent/file_transfer_agent.js:441:14)

Reply:
Could you please check if the last commit fixed this?